File: c/binary_search.c
#include <stdio.h>

// function declaration
int binarySearch(int arr[], int target, int low, int high);

int main(){
	int arr[] = {1,2,3,4,5,6};

	int length = sizeof(arr) / sizeof(arr[0]);
	int result = binarySearch(arr, -88, 0, length - 1);
	
	printf("%d", result);
	return 0;
}

//function declaration
int binarySearch(int arr[], int target, int low, int high){

	while(low <= high){
		int mid = low + (high - low) / 2;
		int guess = arr[mid];

		if(guess == target){
			return mid;
		}

		if(guess < target){
			low = mid + 1;
			continue;
		}

		high = mid - 1;
	}

	return -1;
}

File: php/binary_search.php
<?php
	function binarySearch($arr, $target) {
		$low = 0;
		$high = count($arr) - 1;
		while ($low <= $high) {
			$mid = floor(($high + $low) / 2);
			$guess = $arr[$mid];

			if ($guess == $target) {
				return $mid;
			}

			if ($guess < $target) {
				$low = $mid + 1;
				continue;
			}

			$high = $mid - 1;
		}

		return -1;
	}
?>